packages/orchestration/src/orchestration-client.ts

```diff
@@ -78,31 +202,73 @@ export function constructCompletionPostRequestFromJson(
   }
 }
 
+function addStreamOptions(
+  moduleConfigs: ModuleConfigs,
+  streamOptions?: StreamOptions
+): OrchestrationConfig {
+  const llm = streamOptions?.llm ?? {};
+  const outputFiltering = streamOptions?.outputFiltering ?? {};
+  const chunkSize = streamOptions?.chunkSize;
+
+  return {
+    stream: true,
+    stream_options: {
+      chunk_size: chunkSize
+    },
+    module_configurations: {
+      ...moduleConfigs,
+      llm_module_config: {
+        ...moduleConfigs.llm_module_config,
+        stream_options: {
+          include_usage: llm.includeUsage ?? true,
+          ...llm
+        }
+      },
+      ...(Object.keys(outputFiltering).length && {
+        filtering_module_config: {
+          ...moduleConfigs.filtering_module_config,
+          output: {
+            ...(moduleConfigs.filtering_module_config?.output || {}),
+            stream_options: outputFiltering
+          }
+        }
+      })
+    }
+  } as OrchestrationConfig; // TODO: Remove typecast when types are re-generated;
+}
+
 /**
  * @internal
  */
 export function constructCompletionPostRequest(
   config: OrchestrationModuleConfig,
-  prompt?: Prompt
+  prompt?: Prompt,
+  stream?: boolean,
+  streamOptions?: StreamOptions
 ): CompletionPostRequest {
-  return {
-    orchestration_config: {
-      module_configurations: {
-        templating_module_config: {
-          template: config.templating.template
-        },
-        llm_module_config: config.llm,
-        ...(Object.keys(config?.filtering || {}).length && {
-          filtering_module_config: config.filtering
-        }),
-        ...(Object.keys(config?.masking || {}).length && {
-          masking_module_config: config.masking
-        }),
-        ...(Object.keys(config?.grounding || {}).length && {
-          grounding_module_config: config.grounding
-        })
-      }
+  const moduleConfigurations = {
+    templating_module_config: {
+      template: config.templating.template
     },
+    llm_module_config: config.llm,
+    ...(config?.filtering &&
+      Object.keys(config.filtering).length && {
+        filtering_module_config: config.filtering
+      }),
+    ...(config?.masking &&
+      Object.keys(config.masking).length && {
+        masking_module_config: config.masking
+      }),
+    ...(config?.grounding &&
+      Object.keys(config.grounding).length && {
+        grounding_module_config: config.grounding
+      })
+  };
+
+  return {
+    orchestration_config: stream
+      ? addStreamOptions(moduleConfigurations, streamOptions)
```

Comment:
[pp] As previously mentioned, why don't we allow streamOptions to be optional and get rid of the stream flag.